### ISSUE TITLE
System-notification ads don't auto-dismiss after click/landing, nor do they time-out

### ISSUE THREAD

**Issue by stephendonner** [brave/brave-browser#16357]
## Description 

System-notification ads don't auto-dismiss after click/landing, nor do they time-out

## Steps to Reproduce

   1. new profile
   2. launch Brave using `--enable-logging=stderr --vmodule="*/variations/*"=6,"*/bat-native-ledger/*"=6,"*/brave_rewards/*"=6,"*/bat-native-ads/*"=6,"*/bat-native-confirmations/*"=6,"*/brave_ads/*"=9,"*/brave_user_model/*"=6  --brave-ads-staging --brave-ads-debug --rewards=staging=true,reconcile-interval=3`
   3. click to enable Rewards
   4. open 4+ tabs
   5. wait
   6. do some activity (type, move mouse)
   7. confirm the system-notification ad shows up
   8. click on the ad
   9. wait...

## Actual result:

The system-ad notification never dismisses upon `click`, and doesn't `time-out` either (as others do in 120 seconds?)

clicked ad | waited
-----------|-------
<img width="1680" alt="Screen Shot 2021-06-10 at 3 58 41 PM" src="https://user-images.githubusercontent.com/387249/121607790-85995d80-ca05-11eb-8b08-e4425642efba.png"> | <img width="1678" alt="Screen Shot 2021-06-10 at 4 03 37 PM" src="https://user-images.githubusercontent.com/387249/121607798-89c57b00-ca05-11eb-96db-ebb097f77f23.png">

It's also possible, in this state, to have stacked ad notifications.

stack | stack unfurled
------|---------------
<img width="372" alt="Screen Shot 2021-06-10 at 12 38 24 PM" src="https://user-images.githubusercontent.com/387249/121608083-1708cf80-ca06-11eb-9df7-ad2f15aa31e6.png"> | <img width="382" alt="Screen Shot 2021-06-10 at 12 38 29 PM" src="https://user-images.githubusercontent.com/387249/121608084-17a16600-ca06-11eb-8ef1-3b3570ba4ff8.png">

## Expected result:

Native/system-notification ad should dismiss upon successful `click`/`land`, per @tmancey in Slack thread: https://bravesoftware.slack.com/archives/CJR5902AW/p1623281231116200

## Reproduces how often: 

100%

## Brave version (brave://version info)

Brave | 1.27.57 Chromium: 91.0.4472.101&nbsp;(Official Build)&nbsp;nightly&nbsp;(x86_64)
-- | --
Revision | af52a90bf87030dd1523486a1cd3ae25c5d76c9b-refs/branch-heads/4472@{#1462}
OS | macOS Version 11.4 (Build 20F71)

## Version/Channel Information:

- Can you reproduce this issue with the current release? no
- Can you reproduce this issue with the beta channel? yes
- Can you reproduce this issue with the nightly channel? yes

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? 
- Does the issue resolve itself when disabling Brave Rewards? 
- Is the issue reproducible on the latest version of Chrome? 

## Miscellaneous Information:

/cc @btlechowski

**Comment by stephendonner** [comment]
Also broken in 

Brave | 1.26.59 Chromium: 91.0.4472.101&nbsp;(Official Build)&nbsp;beta&nbsp;(x86_64)
-- | --
Revision | af52a90bf87030dd1523486a1cd3ae25c5d76c9b-refs/branch-heads/4472@{#1462}
OS | macOS Version 11.4 (Build 20F71)

/cc @tmancey

**Comment by btlechowski** [comment]
Not reproducible on Ubuntu


Brave | 1.26.59 Chromium: 91.0.4472.101&nbsp;(Official Build)&nbsp;beta&nbsp;(64-bit)
-- | --
Revision | af52a90bf87030dd1523486a1cd3ae25c5d76c9b-refs/branch-heads/4472@{#1462}
OS | Ubuntu 18.04 LTS



Click:
![16357](https://user-images.githubusercontent.com/34715963/121727392-219d9480-caec-11eb-833e-4629c3050314.gif)

Dismiss:
![16357_2](https://user-images.githubusercontent.com/34715963/121727589-645f6c80-caec-11eb-829f-7320c5efcb7d.gif)

**Comment by LaurenWags** [comment]
Reproduced with

```
Brave | 1.26.58 Chromium: 91.0.4472.88&nbsp;(Official Build)&nbsp;dev&nbsp;(x86_64)
-- | --
Revision | 109e9cd038b94a631aea7d40ee3d56c1278f2597-refs/branch-heads/4472@{#1385}
OS | macOS Version 10.15.7 (Build 19H1030)
```

**Comment by stephendonner** [comment]
**Verified `PASSED`** using the inline steps here on build [1055:259:0618/153723.118473:VERBOSE1:ad_transfer.cc(130)] 

**Steps:**
1. new profile
2. launch Brave using --enable-logging=stderr --vmodule="*/variations/*"=6,"*/bat-native-ledger/*"=6,"*/brave_rewards/*"=6,"*/bat-native-ads/*"=6,"*/bat-native-confirmations/*"=6,"*/brave_ads/*"=9,"*/brave_user_model/*"=6 --brave-ads-staging --brave-ads-debug --rewards=staging=true,reconcile-interval=3
3. click to enable Rewards
4. open 4+ tabs
5. wait
6. do some activity (type, move mouse)
7. confirm the system-notification ad shows up
8. click on the ad

**Confirmed:**
* ad was automatically dismissed, upon `click`
* ad landed `Transferred ad for https://www.brave.com/foo
[1055:259:0618/153723.118889:VERBOSE1:confirmations.cc(90)] Confirming landed ad for creative instance id 2c658159-e18a-4ba7-b4f0-2c51e4eae9ad`
* `click` registered: `[1055:259:0618/153712.615745:VERBOSE1:confirmations.cc(90)] Confirming click ad for creative instance id 2c658159-e18a-4ba7-b4f0-2c51e4eae9ad`
* `click` logged: `[1055:259:0618/153712.658101:VERBOSE1:ad_notification_event_clicked.cc(38)] Successfully logged ad notification clicked event`
* ad showed up on `brave://rewards/`
* BAT shown on new-tab page widget

**Screencast** (please be patient, bit of a wait in the beginning):
![ad-dismiss](https://user-images.githubusercontent.com/387249/122621945-f0711700-d04b-11eb-8388-660bcf08410a.gif)

**Screenshots:**

new-tab page widget | `brave://rewards`
---------|---------
<img width="1459" alt="Screen Shot 2021-06-18 at 3 42 33 PM" src="https://user-images.githubusercontent.com/387249/122621970-0979c800-d04c-11eb-9256-2aa511b7ba83.png"> | <img width="1459" alt="Screen Shot 2021-06-18 at 3 42 49 PM" src="https://user-images.githubusercontent.com/387249/122621971-0a125e80-d04c-11eb-9a38-e28c0c919468.png">
